include/vast/Translation/CodeGenFallBackVisitor.hpp

```diff
@@ -31,21 +32,54 @@ namespace vast::cg
         }
     };
 
-    template< typename Derived >
+    template< typename Derived
+        , template< typename > typename FallBackMixin = UnsupportedFallBackVisitor
+    >
     struct DefaultFallBackVisitorMixin
         : DefaultFallBackDeclVisitorMixin< Derived >
         , DefaultFallBackStmtVisitorMixin< Derived >
         , DefaultFallBackTypeVisitorMixin< Derived >
-    {
+        , FallBackMixin< Derived >
+	{
         using DeclVisitor = DefaultFallBackDeclVisitorMixin< Derived >;
         using StmtVisitor = DefaultFallBackStmtVisitorMixin< Derived >;
         using TypeVisitor = DefaultFallBackTypeVisitorMixin< Derived >;
 
-        using DeclVisitor::Visit;
-        using StmtVisitor::Visit;
-        using TypeVisitor::Visit;
+        using UnsupportedVisitor = FallBackMixin< Derived >;
+
+        Operation* Visit(const clang::Stmt *stmt) {
```

Comment:
I am really confused why we are doing dispatch here again, when a few lines below it is done for us? `CodeGenFallBackVisitorMixin`.

Reply:
Shouldn't the only change to this class be change of those type alias of partial Visitors to the `Unsupported` versions?